component/src/main/java/org/wso2/extension/siddhi/io/http/sink/util/HttpSinkUtil.java

```diff
@@ -0,0 +1,196 @@
+            String[] spam = headers.split(HttpConstants.HEADERSPLITER);
+            for (String aSpam : spam) {
+                String[] header = aSpam.split(HttpConstants.HEADERNAMEVALUESPLITER, 2);
+                if (header.length > 1) {
+                    headersList.add(new Header(header[0], header[1]));
+                } else {
+                    throw new HttpSinkAdaptorRuntimeException(
+                            "Invalid header format. Please include as key1:value1,key2:value2,..");
+                }
+            }
+        }
+        return headersList;
+    }
+
+    /**
+     * user can give custom trust store file and trust store pass if user did not give then custom then system read
+     * the default values which is in the deployment yaml.
+     *
+     * @return default trust store file and pass which is in deployment yml.
+     */
+    public String[] trustStoreValues(ConfigReader sinkConfigReader) {
+        return new String[]{sinkConfigReader.readConfig(HttpConstants.TRUSTSTORE_FILE, HttpConstants
+                .TRUSTSTORE_FILE_VALUE),
+                sinkConfigReader.readConfig(HttpConstants.TRUSTSTORE_PASS, HttpConstants.
+                        TRUSTSTORE_PASS_VALUE)};
+    }
+
+    /**
+     * Method is responsible for set sender configuration values .
+     *
+     * @param httpStaticProperties the map that url details.
+     * @param clientStoreFile      the client trust store file path.
+     * @param clientStorePass      the client trust store pass path.
+     * @return set of sender configurations.
+     */
+    public Set<SenderConfiguration> getSenderConfigurations(Map<String, String> httpStaticProperties, String
+            clientStoreFile, String clientStorePass) {
+        Set<SenderConfiguration> senderConf;
+        if (httpStaticProperties.get(HttpConstants.PROTOCOL).equals(HttpConstants.PROTOCOL_HTTPS)) {
+            SiddhiHttpSenderConfiguration httpsSender = new SiddhiHttpSenderConfiguration(clientStoreFile,
+                    clientStorePass);
+            httpsSender.setId(httpStaticProperties.get(HttpConstants.TO));
+            httpsSender.setScheme(httpStaticProperties.get(HttpConstants.PROTOCOL));
+            senderConf = new HashSet<>(Collections.singletonList(httpsSender));
+        } else {
+            SenderConfiguration httpSender = new SenderConfiguration(httpStaticProperties.get(HttpConstants.TO));
+            httpSender.setScheme(httpStaticProperties.get(HttpConstants.PROTOCOL));
+            senderConf = new HashSet<>(Collections.singletonList(httpSender));
+        }
+        return senderConf;
+    }
+
+    /**
+     * Method is responsible for set transportation configuration values.
+     *
+     * @return return the set of netty transportation configuration.
+     */
+    public Set<TransportProperty> getTransportConfigurations(ConfigReader sourceConfigReader) {
+        ArrayList<TransportProperty> properties = new ArrayList<>();
```

Comment:
Declaration should be a list

Reply:
Fixed in 156b93ab839dfc225e0a915f8951b61db7afa2ab